### PR TITLE
Ternary_Search

### LINKED ISSUE #2176: Ternary Search
- [x] C
- [x] C++
- [x] C# - @p-soham11
- [x] CoffeeScript
- [ ] Dart - @heisastark
- [ ] Go - @Ratna04priya
- [x] Java
- [x] JS
- [ ] Kotlin - @saloniankita23
- [x] Php - @BhardwajShweta25
- [x] Python
- [x] Readme - @shwetagurnani
- [ ] Ruby - @atarax665
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Ratna04priya**
Fixes #2176 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Ternary Search

#### Languages Used:
- Go lang
- 

#### Files Added:
- Ternary_Search.go
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raghav-dalmia** [comment]
Add issue number in the description.

**Comment by raghav-dalmia** [comment]
Also, remove irrelevant files.

**Comment by Ratna04priya** [comment]
I have deleted the irrelevant files. Now it's ready to be merged.

**Comment by somya-kapoor** [comment]
Please add sample input and output. Also, squash the commits.

**Comment by somya-kapoor** [comment]
@Ratna04priya Are you working on it?

**Comment by Ratna04priya** [comment]
Yes, @somya-kapoor. Should I attach the sample input-output files with the main file?

**Comment by somya-kapoor** [comment]
> Yes, @somya-kapoor. Should I attach the sample input-output files with the main file?

Yes

**Comment by Ratna04priya** [comment]
Added the output to the end of the file. @somya-kapoor  Can it be merged now?

File: Ternary_Search/Ternary_Search.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+STEP: 8  l: 7363  r: 7366  slider1: 7364  slider2: 7365                                                                   
+STEP: 9  l: 7363  r: 7363  slider1: 7363  slider2: 7363                                                                   
+-1                                                                                                                        
+                                                                                                                          
+                                                                                                                          
+...Program finished with exit code 0                                                                                      
+Press ENTER to exit console.
+
+*/
+
```
Comment (somya-kapoor) [inline Ternary_Search/Ternary_Search.go:146]: Remove this line

File: Ternary_Search/Ternary_Search.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+			l = slider1 + 1
+			r = slider2 - 1
+		} else {
+			l = slider2 + 1
+		}
+	}
+}
+
+func binarySearch(arr []int, k int) int {
+
```
Comment (somya-kapoor) [inline Ternary_Search/Ternary_Search.go:84]: Use 4 space indentation

File: Ternary_Search/Ternary_Search.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
+package main
```
Comment (somya-kapoor) [inline Ternary_Search/Ternary_Search.go:1]: Use 4 space indentation

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days, I am closing this pr.

**Pull request closed without merging**